GEMele/GEMdiffusion.C

```diff
@@ -21,32 +24,74 @@ int main(int argc,char** argv){
   const float tIntegration = atof(argv[3]);
   const float pitch = atof(argv[4]);
   const int nEvents = atoi(argv[5]);
-  
+  const float field = atof(argv[6]);
+
+
   //____________________________________________________________________________________________________________________________________________________________
   // Gas and detector parameters  
   
+  //Here: TSpline for gas diffusion values depending on field --> for now only txt-file into ntuple
+
   const char *gas="Ne-CO2-N2_90-10-5";
   float wIon = 37.3f; float diffL = 0.0218f; float diffT = 0.0223f; float vdrift = 2.52f;
-  if(gasFlag == 1) {gas = "Ne-CO2_90-10";  wIon = 38.1f; diffL = 0.0223f; diffT = 0.0219f; vdrift = 2.66f; }             
-  if(gasFlag == 2) {gas = "Ar-CO2_90-10";  wIon = 28.8f; diffL = 0.0244f; diffT = 0.0268f; vdrift = 3.25f; }
+  if(gasFlag == 1) {gas = "Ne-CO2_90-10";  wIon = 38.1f; diffL = 0.0223f; diffT = 0.0219f; vdrift = 2.66f; }  
+//  if(gasFlag == 2) {gas = "Ar-CO2_90-10";  wIon = 28.8f; diffL = 0.0244f; diffT = 0.0268f; vdrift = 3.25f; }
+
+  if(gasFlag ==2) {
+
+    gas = "Ar-CO2_90-10"; 
+    wIon = 28.8f;
+
+    FILE *gasdat = fopen("/home/ga24get/Ar_90_CO2_10_fullv.dat","r");
+
+    float field_A, diffL_A, diffT_A, vdrift_A;
+    int nlines = 0;
+    int ncols;
+
+    TNtuple *ntuple = new TNtuple("ntuple","gas transport coeff","field_A:diffL_A:diffT_A:vdrift_A");
+
+    while(1) {
+      ncols = fscanf(gasdat,"%f %f %f %f", &field_A, &diffL_A, &diffT_A, &vdrift_A);
+      if (ncols < 0) break;
+      ntuple->Fill(field_A, diffL_A, diffT_A, vdrift_A);
+      ++nlines;	
+    }
+
+    bool isField = false;
+    for (int row=0; row < ntuple->GetEntries(); row++) {
+      ntuple->GetEntry(row);
+      if (std::abs(ntuple->GetArgs()[0] - field) < 0.01f) { 
+         isField = true;
+         diffL= ntuple->GetArgs()[2]/10000.f;
+         diffT = ntuple->GetArgs()[3]/10000.f;
+         vdrift = ntuple->GetArgs()[1];
```

Comment:
Why not do the comparison when you loop over the file itself? There is actually no need to cast this into the TNtuple and do the comparison there